### PR TITLE
feat: HS-148: Added Surcharge for One Click Wallets

### PR DESCRIPTION
**Pull request by ArushKapoorJuspay**
- Added Surcharge For One Click Wallets
- Added Surcharge For missed payment methods

Screenshot 2024-01-18 at 8 38 47 PM

**Improved UI**

Screenshot 2024-01-19 at 3 16 37 PM

### CODE REVIEW

File: src/Components/SurchargeUtils.res [hunk 1]
```diff
@@ -0,0 +1,89 @@
+type oneClickWallets = {
+  paymentMethodType: string,
+  displayName: string,
+}
+let oneClickWallets = [
+  {paymentMethodType: "apple_pay", displayName: "ApplePay"},
+  {paymentMethodType: "paypal", displayName: "Paypal"},
+  {paymentMethodType: "google_pay", displayName: "GooglePay"},
```
Comment (akash-kamble-juspay) [inline src/Components/SurchargeUtils.res:8]: @ArushKapoorJuspay @prafulkoppalkar is this an exhaustive list of one-click payment methods?

File: src/LocaleString.res [hunk 2]
```diff
@@ -156,9 +162,12 @@ let localeStrings = [
     surchargeMsgAmount: str => `A surcharge amount of ${str} will be applied for this transaction`,
     surchargeMsgAmountForCard: str =>
       `A surcharge amount of upto ${str} will be applied for this transaction`,
+    surchargeMsgAmountForOneClickWallets: "Additional fee applicable",
```
Comment (akash-kamble-juspay) [inline src/LocaleString.res:165]: @prafulkoppalkar you had made a change wherein if the once click payment methods need additional dynamic fields (like state/ country), the payment method is shown besides card, and is no more one click. How is Surcharge handled for that?

Comment (prafulkoppalkar) [reply src/LocaleString.res:179]: @ArushKapoorJuspay has handled that .. it will come below similar to other payment methods surcharge

File: src/Utilities/Utils.res [hunk 3]
```diff
@@ -835,3 +835,5 @@ let isOtherElements = componentType => {
   componentType == "cardExpiry" ||
   componentType == "cardCvc"
 }
+
+let nbsp = `\u00A0`
```
Comment (prafulkoppalkar) [inline src/Utilities/Utils.res:839]: why is this needed ? using this is not encouraged

File: src/Components/SurchargeUtils.res (hunk 1 above)
Comment (prafulkoppalkar) [reply src/Components/SurchargeUtils.res:8]: Yes, it is @ArushKapoorJuspay only these methods can be one click checkouts right ?

**Review by prafulkoppalkar — APPROVED** [review]

File: src/Utilities/Utils.res (hunk 3 above)
Comment (ArushKapoorJuspay) [reply src/Utilities/Utils.res:839]: This is for adding space in string, if I directly use React.string(` some `), it will trim the spaces and show some

**Review by akash-kamble-juspay — APPROVED** [review]
